frontend/src/app/modules/result-selection/services/result-selection.store.ts

```diff
@@ -165,6 +165,7 @@ export class ResultSelectionStore {
     this.userTimings$.next({...this.userTimings$.getValue(), isLoading: true});
     this.resultSelectionService.fetchResultSelectionData<SelectableMeasurand[]>(resultSelectionCommand, URL.USER_TIMINGS)
       .subscribe(next => {
+      next.map(userTiming => userTiming.kind = "selectable-measurand");
```

Comment:
You are mis-using map function here. Map should be used to map one data structure in to another (new) one, leaving the original data structure unchanged. You are changing attributes `kind` of objects in the original list as a side effect while discarding the result of the map function itself (which is actually `['selectable-measurand', 'selectable-measurand', 'selectable-measurand', ...]`.

Better explicitly change objects with forEach:
```suggestion
      next.forEach((userTiming: SelectableMeasurand) => userTiming.kind = 'selectable-measurand');
```

Reply:
Or one could use map to build new `SelectableMeasurands` with kind attribute:
```
let responseWithLoadingState: MeasurandGroup = {
        isLoading: false,
        name: groupName,
        values: next.map((userTiming: SelectableMeasurand) => { return {...userTiming, kind: 'selectable-measurand'}})
      };
```
Therefore type of `kind` attribute must be changed from unique types `"selectable-measurand"` / `"performance-aspect-type"` to type string in `SelectableMeasurand` and `PerformanceAspectType`.